packages/playwright-core/src/server/dispatchers/dispatcher.ts

```diff
@@ -357,7 +357,7 @@ export class DispatcherConnection {
     }
 
     if (response.error)
-      response.log = callMetadata.log;
+      response.log = compressCallLog(callMetadata.log);
```

Comment:
I expect a similar change in `FrameDispatcher::expect` which would allow us to remove all of these, then the language ports will benefit from your repeating sub-sequence algorithm as well:

<img width="621" alt="image" src="https://github.com/user-attachments/assets/ca6c027e-ddd5-488a-b13a-903cad09cd00">

Reply:
log is already compressed here since it comes from the Frame.expect protocol call.